files/query_cache.py

```
@@ -0,0 +1,424 @@
+    """
+    try:
+        version_key = _get_cache_version_key(scope, identifier)
+
+        # Try atomic increment first (Redis, Memcached)
+        try:
+            new_version = cache.incr(version_key)
+            # Update the key with the new TTL using Django's cache.set()
+            # This ensures the version key doesn't expire prematurely
+            cache.set(version_key, new_version, CACHE_VERSION_TIMEOUT)
+            logger.debug(f"Bumped cache version for {scope}:{identifier} to {new_version}")
+            return new_version
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)